### DISCUSSION TITLE
Unit of morb and Morb

### DISCUSSION THREAD

**Discussion by yilinzhao222** [wannier-berri/wannier-berri#301] [Q&A]
Hi Stepan,

Sorry to bother you with the unit issue again. I noticed that the unit of Morb (wberri.calculators.static.Morb) is '\mu_B'. Is the unit of morb (wberri.calculators.tabulate.OrbitalMoment) '\mu_B' or SI unit 'A*m^2'?

In addition, I get kind of confused when calculating morb with w90 system and pythTB systems. Could you give me a clue about the formula used to calculate the morb (wberri.calculators.tabulate.OrbitalMoment) please? Is there any difference for the formula of morb between these two kinds of systems?

Best regards,
Yilin

**Comment by stepan-tsirkin** [comment]
The `morb` trabulator evaluates the `morb` formula 
https://github.com/wannier-berri/wannier-berri/blob/3ee1fb0c4b3f751613fefac80f9ae71e074a034c/wannierberri/formula/covariant.py#L338-L341
Which is $\langle \nabla_{\mathbf{k}} u_m | \times \left(H-\frac{E_m+E_n}{2}\right)| \nabla_{\mathbf{k}} u_m \rangle$, so the pre-factors are dropped down, and the units are eV*Ang^2 (internally the units of energies are eV and units of length are Ang)

The formula is independent of the system, just in a tight-binding model one does not evaluate the "external terms".
